packages/noya-state/src/reducers/canvasReducer.ts

```diff
@@ -264,13 +274,77 @@ export function canvasReducer(
         };
       });
     }
+    case 'moveLayersIntoParentAtPoint': {
+      const [, , point] = action;
+
+      const pageIndex = getCurrentPageIndex(state);
+      const indexPaths = getSelectedLayerIndexPathsExcludingDescendants(state);
+
+      const parentIndexPath = Layers.findIndexPath(
+        state.sketch.pages[pageIndex],
+        (layer) =>
+          (Layers.isArtboard(layer) || Layers.isSymbolMaster(layer)) &&
+          rectContainsPoint(layer.frame, point),
+      );
+
+      return produce(state, (draft) => {
```

Comment:
After thinking about it some more, I think we should try to re-use `'moveLayer'` here. This code is kinda tricky, so it's better if we only have it in one place. Let's make a new function in `layerReducer`:

```ts
export function moveLayer(
  state: ApplicationState,
  id: string | string[],
  destinationId: string,
  rawPosition: RelativeDropPosition,
) {
  // ...
  return produce(state, draft => { ... })
}
```

And move the [entire 'moveLayer' case](https://github.com/noya-app/noya/blob/f60c0ef1af834765386ea9fc649aa3b73e8f7a8b/packages/noya-state/src/reducers/layerReducer.ts#L197-L285) into that function.

Then, here within `'moveLayersIntoParentAtPoint'`, we can call it:

```ts
const parentId = mouseIsInParentLayer(page, point)?.do_objectID ?? page.do_objectID

return moveLayer(state, state.selectedObjects, parentId, 'inside');
```

If there's any behavior that's special to this dragging use case, we could make it another param of `moveLayer()`, or handle it before calling `moveLayer()`

Reply:
the move layer function has a "position" variable that I don't need in the other drag function, and it's used right in the middle of the moveLayer function. Do I remove it ?🤔

Reply:
Oh I see how it's supposed to work

Reply:
okey I see it now 😆 I think all good